plugins/inputs/opentelemetry_listener/grpc_services.go

```diff
@@ -0,0 +1,72 @@
+package opentelemetry_listener
+
+import (
+	"context"
+
+	"github.com/influxdata/influxdb-observability/otel2influx"
+	otlpcollectorlogs "github.com/influxdata/influxdb-observability/otlp/collector/logs/v1"
+	otlpcollectormetrics "github.com/influxdata/influxdb-observability/otlp/collector/metrics/v1"
+	otlpcollectortrace "github.com/influxdata/influxdb-observability/otlp/collector/trace/v1"
+)
+
+type grpcServiceBase struct {
+	influxConverter *otel2influx.OpenTelemetryToInfluxConverter
+	influxWriter    *writer
+}
+
+type traceService struct {
+	otlpcollectortrace.UnimplementedTraceServiceServer
+	grpcServiceBase
+}
+
+func newTraceService(influxConverter *otel2influx.OpenTelemetryToInfluxConverter, influxWriter *writer) *traceService {
```

Comment:
Do we really need to pass the converter and writer in here? I think you can just make this a member function of the plugin struct and construct the instances in here?

Reply:
Ok after reading the main plugin code, it might make sense to keep this as-is... You decide, if you want to keep is simply close the conversation as resolved. :-)